[inductor] Update cudagraphs partition logging

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #170218
* __->__ #170217

Summary: Make TORCH_LOGS=cudagraphs dump more information about the partitions, such as why non-cudagraphable nodes are non-cudagraphable. Here is an example of the new log,

```
torch/_inductor/scheduler.py:6005] [0/1] [__cudagraphs] Created 4 graph partitions: 2 cudagraphable, 2 non-cudagraphable
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 0: 6 nodes, non-cudagraphable, inputs=0, outputs=0
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op0: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op1: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op2: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op3: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op4: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op5: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 1: 2 nodes, cudagraphable, inputs=2, outputs=4
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 2: 3 nodes, non-cudagraphable, inputs=1, outputs=1
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op8: reason=cpu ops, ir=DeviceCopy, fx=prims.device_put.default(add_1, cpu)
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]          File "test/inductor/test_cudagraph_trees.py", line 4032, in f
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]              y_cpu = y1.cpu() + 1
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op9: reason=cpu ops, ir=ComputedBuffer
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op10: reason=DeviceCopy ops, ir=DeviceCopy, fx=prims.device_put.default(add_2, cuda:0)
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]          File "test/inductor/test_cudagraph_trees.py", line 4035, in f
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]              return x1 + y1 + z + y_cpu.to("cuda")
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 3: 1 nodes, cudagraphable, inputs=4, outputs=1
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @Lucaskabela @mlazos

Differential Revision: [D88980119](https://our.internmc.facebook.com/intern/diff/D88980119)

desertfire committed:
[inductor] Update cudagraphs partition logging

Summary: Make TORCH_LOGS=cudagraphs dump more information about the partitions, such as why non-cudagraphable nodes are non-cudagraphable. Here is an example of the new log,

```
torch/_inductor/scheduler.py:6005] [0/1] [__cudagraphs] Created 4 graph partitions: 2 cudagraphable, 2 non-cudagraphable
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 0: 6 nodes, non-cudagraphable, inputs=0, outputs=0
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op0: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op1: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op2: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op3: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op4: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op5: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 1: 2 nodes, cudagraphable, inputs=2, outputs=4
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 2: 3 nodes, non-cudagraphable, inputs=1, outputs=1
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op8: reason=cpu ops, ir=DeviceCopy, fx=prims.device_put.default(add_1, cpu)
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]          File "test/inductor/test_cudagraph_trees.py", line 4032, in f
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]              y_cpu = y1.cpu() + 1
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op9: reason=cpu ops, ir=ComputedBuffer
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op10: reason=DeviceCopy ops, ir=DeviceCopy, fx=prims.device_put.default(add_2, cuda:0)
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]          File "test/inductor/test_cudagraph_trees.py", line 4035, in f
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]              return x1 + y1 + z + y_cpu.to("cuda")
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 3: 1 nodes, cudagraphable, inputs=4, outputs=1
```

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D88980119).

desertfire committed:
Update on "[inductor] Update cudagraphs partition logging"


Summary: Make TORCH_LOGS=cudagraphs dump more information about the partitions, such as why non-cudagraphable nodes are non-cudagraphable. Here is an example of the new log,

```
torch/_inductor/scheduler.py:6005] [0/1] [__cudagraphs] Created 4 graph partitions: 2 cudagraphable, 2 non-cudagraphable
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 0: 6 nodes, non-cudagraphable, inputs=0, outputs=0
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op0: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op1: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op2: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op3: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op4: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op5: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 1: 2 nodes, cudagraphable, inputs=2, outputs=4
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 2: 3 nodes, non-cudagraphable, inputs=1, outputs=1
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op8: reason=cpu ops, ir=DeviceCopy, fx=prims.device_put.default(add_1, cpu)
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]          File "test/inductor/test_cudagraph_trees.py", line 4032, in f
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]              y_cpu = y1.cpu() + 1
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op9: reason=cpu ops, ir=ComputedBuffer
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op10: reason=DeviceCopy ops, ir=DeviceCopy, fx=prims.device_put.default(add_2, cuda:0)
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]          File "test/inductor/test_cudagraph_trees.py", line 4035, in f
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]              return x1 + y1 + z + y_cpu.to("cuda")
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 3: 1 nodes, cudagraphable, inputs=4, outputs=1
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo mlazos

Differential Revision: [D88980119](https://our.internmc.facebook.com/intern/diff/D88980119)

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D88980119).

eellison reviewed (commented):
would you mind waiting for https://github.com/pytorch/pytorch/pull/170103 to land, this also includes fix and we need it for 2.10

BoyuanFeng commented on torch/_inductor/scheduler.py:
```diff
@@ -5493,42 +5494,33 @@ def should_partition(
-            return
-
-        log_partition_reason = maybe_log_cudagraph_partition if should_log else noop_log
+            return "cudagraphs disabled"
```
maybe elaborate the reason here to avoid confusion?
```suggestion
            return "partition on all ops when cudagraphs is disabled"
```

desertfire committed:
Update on "[inductor] Update cudagraphs partition logging"


Summary: Make TORCH_LOGS=cudagraphs dump more information about the partitions, such as why non-cudagraphable nodes are non-cudagraphable. Here is an example of the new log,

```
torch/_inductor/scheduler.py:6005] [0/1] [__cudagraphs] Created 4 graph partitions: 2 cudagraphable, 2 non-cudagraphable
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 0: 6 nodes, non-cudagraphable, inputs=0, outputs=0
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op0: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op1: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op2: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op3: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op4: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op5: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 1: 2 nodes, cudagraphable, inputs=2, outputs=4
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 2: 3 nodes, non-cudagraphable, inputs=1, outputs=1
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op8: reason=cpu ops, ir=DeviceCopy, fx=prims.device_put.default(add_1, cpu)
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]          File "test/inductor/test_cudagraph_trees.py", line 4032, in f
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]              y_cpu = y1.cpu() + 1
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op9: reason=cpu ops, ir=ComputedBuffer
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op10: reason=DeviceCopy ops, ir=DeviceCopy, fx=prims.device_put.default(add_2, cuda:0)
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]          File "test/inductor/test_cudagraph_trees.py", line 4035, in f
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]              return x1 + y1 + z + y_cpu.to("cuda")
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 3: 1 nodes, cudagraphable, inputs=4, outputs=1
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo Lucaskabela mlazos

Differential Revision: [D88980119](https://our.internmc.facebook.com/intern/diff/D88980119)

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D88980119).

desertfire committed:
Update on "[inductor] Update cudagraphs partition logging"


Summary: Make TORCH_LOGS=cudagraphs dump more information about the partitions, such as why non-cudagraphable nodes are non-cudagraphable. Here is an example of the new log,

```
torch/_inductor/scheduler.py:6005] [0/1] [__cudagraphs] Created 4 graph partitions: 2 cudagraphable, 2 non-cudagraphable
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 0: 6 nodes, non-cudagraphable, inputs=0, outputs=0
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op0: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op1: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op2: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op3: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op4: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op5: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 1: 2 nodes, cudagraphable, inputs=2, outputs=4
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 2: 3 nodes, non-cudagraphable, inputs=1, outputs=1
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op8: reason=cpu ops, ir=DeviceCopy, fx=prims.device_put.default(add_1, cpu)
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]          File "test/inductor/test_cudagraph_trees.py", line 4032, in f
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]              y_cpu = y1.cpu() + 1
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op9: reason=cpu ops, ir=ComputedBuffer
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op10: reason=DeviceCopy ops, ir=DeviceCopy, fx=prims.device_put.default(add_2, cuda:0)
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]          File "test/inductor/test_cudagraph_trees.py", line 4035, in f
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]              return x1 + y1 + z + y_cpu.to("cuda")
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 3: 1 nodes, cudagraphable, inputs=4, outputs=1
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo Lucaskabela mlazos

Differential Revision: [D88980119](https://our.internmc.facebook.com/intern/diff/D88980119)

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D88980119).

desertfire committed:
Update on "[inductor] Update cudagraphs partition logging"


Summary: Make TORCH_LOGS=cudagraphs dump more information about the partitions, such as why non-cudagraphable nodes are non-cudagraphable. Here is an example of the new log,

```
torch/_inductor/scheduler.py:6005] [0/1] [__cudagraphs] Created 4 graph partitions: 2 cudagraphable, 2 non-cudagraphable
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 0: 6 nodes, non-cudagraphable, inputs=0, outputs=0
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op0: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op1: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op2: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op3: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op4: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op5: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 1: 2 nodes, cudagraphable, inputs=2, outputs=4
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 2: 3 nodes, non-cudagraphable, inputs=1, outputs=1
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op8: reason=cpu ops, ir=DeviceCopy, fx=prims.device_put.default(add_1, cpu)
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]          File "test/inductor/test_cudagraph_trees.py", line 4032, in f
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]              y_cpu = y1.cpu() + 1
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op9: reason=cpu ops, ir=ComputedBuffer
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op10: reason=DeviceCopy ops, ir=DeviceCopy, fx=prims.device_put.default(add_2, cuda:0)
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]          File "test/inductor/test_cudagraph_trees.py", line 4035, in f
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]              return x1 + y1 + z + y_cpu.to("cuda")
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 3: 1 nodes, cudagraphable, inputs=4, outputs=1
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo Lucaskabela mlazos

Differential Revision: [D88980119](https://our.internmc.facebook.com/intern/diff/D88980119)

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Update cudagraphs partition logging"


Summary: Make TORCH_LOGS=cudagraphs dump more information about the partitions, such as why non-cudagraphable nodes are non-cudagraphable. Here is an example of the new log,

```
torch/_inductor/scheduler.py:6005] [0/1] [__cudagraphs] Created 4 graph partitions: 2 cudagraphable, 2 non-cudagraphable
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 0: 6 nodes, non-cudagraphable, inputs=0, outputs=0
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op0: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op1: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op2: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op3: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op4: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op5: reason=cpu ops, ir=AssertScalar
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 1: 2 nodes, cudagraphable, inputs=2, outputs=4
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 2: 3 nodes, non-cudagraphable, inputs=1, outputs=1
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op8: reason=cpu ops, ir=DeviceCopy, fx=prims.device_put.default(add_1, cpu)
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]          File "test/inductor/test_cudagraph_trees.py", line 4032, in f
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]              y_cpu = y1.cpu() + 1
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op9: reason=cpu ops, ir=ComputedBuffer
torch/_inductor/scheduler.py:6040] [0/1] [__cudagraphs]     op10: reason=DeviceCopy ops, ir=DeviceCopy, fx=prims.device_put.default(add_2, cuda:0)
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]          File "test/inductor/test_cudagraph_trees.py", line 4035, in f
torch/_inductor/scheduler.py:6047] [0/1] [__cudagraphs]              return x1 + y1 + z + y_cpu.to("cuda")
torch/_inductor/scheduler.py:6012] [0/1] [__cudagraphs]   Partition 3: 1 nodes, cudagraphable, inputs=4, outputs=1
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo Lucaskabela mlazos

Differential Revision: [D88980119](https://our.internmc.facebook.com/intern/diff/D88980119)

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D88980119).
